client/src/pages/locations/Form.js

```diff
@@ -111,6 +123,14 @@ const LocationForm = ({
       }
     }
   }
+  const checkPotentialDuplicatesDebounced = useDebouncedCallback(
+    checkPotentialDuplicates,
+    400
+  )
+  const [locationName, setLocationName] = useState(initialValues?.name)
+  useEffect(() => {
+    checkPotentialDuplicatesDebounced(locationName)
+  }, [locationName, checkPotentialDuplicatesDebounced])
```

Comment:
Maybe sort these effect dependencies alphabetically (which also happes to be the order in which they appear in the effect itself)?